### ISSUE TITLE
[a11y]: Inline CodeSnippet copy to clipboard notification is not read by JAWS

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13672]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

1.27.0

### React version

v18

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

[Reported internally](https://ibm-studios.slack.com/archives/C054WK4ES5S/p1682510942552969), when testing the inline code snippet with JAWS,

> After activating the "Copy to clipboard" button, The content is copied. However there is no feedback provided by SR. Hence the user may wonder whether the content copied or not.
>
> SR is expected to notify the user about the success of the copy function instantaneously
>
> When there is no feedback of an action that is triggered, Screen reader users would not know whether the action took place or not.

### WCAG 2.1 Violation

4.1.3 Status Messages

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-codesnippet--inline

### Steps to reproduce

Load the story and use JAWS to see how the "copied to clipboard" notification is not read.

**Comment by andreancardona** [comment]
This issue is fixed as seen in this recording https://github.com/carbon-design-system/carbon/pull/16022
